lib/sudo-exec/src/parent.rs

```diff
@@ -33,8 +35,9 @@ impl ParentClosure {
         let mut event_handler = EventHandler::<Self>::new()?;
 
         event_handler.set_read_callback(&backchannel, |ec, ev| ec.check_backchannel(ev));
+        event_handler.set_write_callback(&backchannel, |ec, ev| ec.check_queue(ev));
```

Comment:
if this is a write callback when it does it fire? *not* when there's a message pending to be read from `MonitorClosure` but rather when the `ParentClosure` sends a message to the monitor?

Reply:
It fires when the `backchannel` socket can be written without blocking. That's why we buffer the messages in a `VecDeque` meanwhile. To be honest this shouldn't be an issue most of the time as signals are scarce but better be on the safe side I guess.